src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -102,6 +109,40 @@ public Session authenticate(X x) {
       return null;
     }
 
+    //
+    // Support for Basic HTTP Authentication
+    // Redimentary testing: curl --user username:password http://localhost:8080/service/dig
+    //   visually inspect results, on failure you'll see the dig login page.
+    //
+    if ( ! SafetyUtil.isEmpty(authHeader) ) {
+      StringTokenizer st = new StringTokenizer(authHeader);
+      if (st.hasMoreTokens()) {
```

Comment:
Follow the FOAM coding standard: https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md